From: Johannes Thumshirn <jthumshirn@suse.de>
Subject: KABI protect structy ata_host
Date: Mon Aug  6 11:04:01 CEST 2018
Patch-mainline: Never, KABI fix
References: git-fixes

Patch patches.fixes/libata-add-refcounting-to-ata_host.patch introduced a
kref into struct ata_host. Guard it with a ifndef __GENKSYMS__ and move it
to the end of the structure.

Keeping it at it's current position is unsafe because already built KMPs
expect the 'struct mutex eh_mutex' at this position.

Signed-off-by: Johannes Thumshirn <jthumshirn@suse.de>
---
 include/linux/libata.h |    5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

--- a/include/linux/libata.h
+++ b/include/linux/libata.h
@@ -617,13 +617,16 @@ struct ata_host {
 	void			*private_data;
 	struct ata_port_operations *ops;
 	unsigned long		flags;
-	struct kref		kref;
 
 	struct mutex		eh_mutex;
 	struct task_struct	*eh_owner;
 
 	struct ata_port		*simplex_claimed;	/* channel owning the DMA */
 	struct ata_port		*ports[0];
+#ifndef __GENKSYMS__
+	struct kref		kref;
+#endif
+
 };
 
 struct ata_queued_cmd {
